src/SparkVault.sol

```diff
@@ -266,25 +266,31 @@ contract SparkVault is AccessControlEnumerableUpgradeable, UUPSUpgradeable, ISpa
     /**********************************************************************************************/
 
     function deposit(uint256 assets, address receiver) public returns (uint256 shares) {
+        require(!hasRole(TAKER_ROLE, msg.sender), "SparkVault/taker-cannot-deposit");
+        require(!hasRole(TAKER_ROLE, receiver),   "SparkVault/taker-cannot-be-receiver");
```

Comment:
Lets just put these in `_mint` and use `SparkVault/taker-cannot-deposit`, will be cleaner, can also remove comments in deposit and mint